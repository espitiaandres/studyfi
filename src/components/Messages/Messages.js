//
//  Messages.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import ScrollToBottom from 'react-scroll-to-bottom';
import Message from '../Message/Message';
import './Messages.css';

const Messages = ({ messages, name, season }) => (
    <ScrollToBottom className="messages">
        {messages
            .filter((a, b, c) => c.indexOf(a) === b)
            .map((message, i) => 
                <div key={i}>
                    <Message message={message} name={name} season={season}/>
                </div>
            )
        }
    </ScrollToBottom>
)

export default Messages;
