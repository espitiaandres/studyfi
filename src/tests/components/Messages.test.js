//
//  Messages.test.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import Messages from '../../components/Messages/Messages';

const messagesPropsTest = {
    messages: [{
        user: "testUser",
        text: "This is my message.",
        currentTime: moment().format("HH MM"),
        room: '1'
    }],
    name: "testUser",
    season: true
}

test('should render Messages component correctly', () => {
    const wrapper = shallow(
        <Messages
            messages={messagesPropsTest.messages}
            name={messagesPropsTest.name}
            season={messagesPropsTest.season}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
