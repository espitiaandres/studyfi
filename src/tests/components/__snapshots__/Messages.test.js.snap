// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Messages component correctly 1`] = `ShallowWrapper {}`;
